unity-renderer/Assets/DCLPlugins/ECS7/Systems/SceneBoundsCheckerSystem/ECSSceneBoundsCheckerSystem.cs

```diff
@@ -0,0 +1,358 @@
+            IInternalECSComponent<InternalVisibility> visibilityComponent,
+            IInternalECSComponent<InternalRenderers> renderersComponent,
+            IInternalECSComponent<InternalColliders> pointerColliderComponent,
+            IInternalECSComponent<InternalColliders> physicsColliderComponent,
+            IInternalECSComponent<InternalAudioSource> audioSourceComponent,
+            bool previewMode = false)
+        {
+            this.sceneBoundsCheckComponent = sbcComponent;
+            this.visibilityComponent = visibilityComponent;
+            this.renderersComponent = renderersComponent;
+            this.pointerCollidersComponent = pointerColliderComponent;
+            this.physicsCollidersComponent = physicsColliderComponent;
+            this.audioSourceComponent = audioSourceComponent;
+
+            outOfBoundsVisualFeedback = previewMode ? new ECSOutOfSceneBoundsFeedback_RedWireframe() : new ECSOutOfSceneBoundsFeedback_EnabledToggle();
+        }
+
+        public void Update()
+        {
+            ProcessRendererComponentChanges(sceneBoundsCheckComponent, renderersComponent.GetForAll());
+            ProcessPhysicColliderComponentChanges(sceneBoundsCheckComponent, physicsCollidersComponent.GetForAll());
+            ProcessPointerColliderComponentChanges(sceneBoundsCheckComponent, pointerCollidersComponent.GetForAll());
+            ProcessAudioSourceComponentChanges(sceneBoundsCheckComponent, audioSourceComponent.GetForAll());
+
+            // Note: the components are traversed backwards as we may free the 'fully defaulted' entities from the component
+            var sbcComponentGroup = sceneBoundsCheckComponent.GetForAll();
+
+            for (int i = sbcComponentGroup.Count - 1; i >= 0; i--)
+            {
+                var componentData = sbcComponentGroup[i].value;
+                IParcelScene scene = componentData.scene;
+                IDCLEntity entity = componentData.entity;
+                InternalSceneBoundsCheck model = componentData.model;
+
+                if (!model.dirty || scene.isPersistent) continue;
+
+                bool physicsColliderRemoved = WereColliderComponentRemoved(scene, entity,
+                    model.physicsColliders, physicsCollidersComponent);
+
+                bool pointerColliderRemoved = WereColliderComponentRemoved(scene, entity,
+                    model.pointerColliders, pointerCollidersComponent);
+
+                bool renderersRemoved = WereRendererComponentRemoved(scene, entity,
+                    model.renderers, renderersComponent);
+
+                bool isMeshDirty = model.meshesDirty | physicsColliderRemoved | pointerColliderRemoved | renderersRemoved;
+
+                if (isMeshDirty)
+                {
+                    if (physicsColliderRemoved)
+                        model.physicsColliders = null;
+
+                    if (pointerColliderRemoved)
+                        model.pointerColliders = null;
```

Comment:
Not sure how it's implemented in ECS7, but in one of our spikes for the shader based SBC we thought about removing colliders from this service. You dont actually need collider for scene "A" if you are in scene "B".

You would improve performance by reducing stress to the physics system, and prevent missbehaviours not cached by the SBC. Also the SBC would be more efficient because it can focus only in the rendering.

Reply:
Is this easy to do in the new SDK implementation? Do you find any downside?

Reply:
You mean like directly disabling colliders when user get out of scene?
sdk7 let you click on stuff from other scene, so we probably can't apply something like that for pointer colliders, but it might be a good idea for physics colliders.
we'll keep it in mind to experiment with that approach in the future and compare the performance
Thanks!